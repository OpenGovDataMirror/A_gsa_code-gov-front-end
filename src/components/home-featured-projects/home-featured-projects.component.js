import React, { Component } from 'react'
import HomeFeaturedProject from '../home-featured-project'

export default class HomeFeaturedProjects extends Component {
  /* static propTypes = {
        onChange: PropTypes.func.isRequired,
        onSubmit: PropTypes.func.isRequired,
        placeholder: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
  } */

  render() {
    return (
      <section className="featured-projects block block--white">
        <div id="featured-projects-title">
          <div className="indented">
            <h2>Featured Projects</h2>
          </div>
        </div>
        <div>
          {this.props.featuredProjects &&
            this.props.featuredProjects.map((project, index) => (
              <HomeFeaturedProject index={index} key={project.short_name} project={project} />
            ))}
        </div>
      </section>
    )
  }
}
