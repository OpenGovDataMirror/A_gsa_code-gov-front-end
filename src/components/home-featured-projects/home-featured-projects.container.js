import { connect } from 'react-redux'
import { getConfigValue } from 'utils/other'
import HomeFeaturedProjects from './home-featured-projects.component'

export const mapStateToProps = () => ({
  featuredProjects: getConfigValue('content.home.featured')
})

export default connect(mapStateToProps)(HomeFeaturedProjects)
