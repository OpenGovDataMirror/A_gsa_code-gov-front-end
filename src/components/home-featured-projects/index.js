import FeaturedProjectsContainer from './home-featured-projects.container'

export default FeaturedProjectsContainer
