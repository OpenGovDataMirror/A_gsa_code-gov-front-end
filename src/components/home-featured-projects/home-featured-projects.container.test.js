import { mapStateToProps } from 'components/home-featured-projects/home-featured-projects.container'

describe('containers - HomeFeaturedProjects', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps()).toMatchSnapshot()
    })
  })
})
