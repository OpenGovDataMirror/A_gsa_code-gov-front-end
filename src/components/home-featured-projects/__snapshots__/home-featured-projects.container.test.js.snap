// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomeFeaturedProjects mapStateToProps should return the correct properties 1`] = `
Object {
  "featuredProjects": Array [
    Object {
      "alt": "A network of pipes with a red, yellow and green valve and a meter gauge",
      "author": "Environmental Protection Agency",
      "description": "A python package designed to simulate and analyze resilience of water distribution networks.",
      "image": "/assets/img/featured-projects/pipes4.png",
      "links": Array [
        Object {
          "name": "View Project",
          "url": "/projects/doe_sandia_national_laboratories_snl_water_network_tool_for_resilience_v_1_0",
        },
        Object {
          "name": "Go to Repo",
          "url": "https://github.com/usepa/wntr",
        },
      ],
      "short_name": "WNTR",
      "verbose_name": "Water Network Tool for Resilience",
    },
    Object {
      "alt": "Cartoon truck navigating windy road with caution sign and wireless towers",
      "author": "Department of Transportation",
      "description": "A real-time virtual data router that ingests and processes operational data from various connected devices including vehicles, infrastructure, and traffic management centers and distributes it to other devices and subscribing transportation management applications.",
      "image": "/assets/img/featured-projects/smartcity4.png",
      "links": Array [
        Object {
          "name": "View Project",
          "url": "/projects/dot_fhwa_intelligent_transportation_systems_operational_data_environment_its_ode_",
        },
        Object {
          "name": "Go to Repo",
          "url": "https://github.com/usdot-jpo-ode/jpo-ode",
        },
      ],
      "short_name": "ITS ODE",
      "verbose_name": "Intelligent Transportation Systems Operational Data Environment",
    },
    Object {
      "alt": "Cartoon hand pressing buttons connected to phone and laptop",
      "author": "National Security Agency",
      "description": "Enables security teams to integrate apps, workflows, and analytics tools that would otherwise need to be created from scratch.",
      "image": "/assets/img/featured-projects/WALKOFF.png",
      "links": Array [
        Object {
          "name": "View Project",
          "url": "/projects/nsa_walkoff",
        },
        Object {
          "name": "Go to Repo",
          "url": "https://github.com/iadgov/WALKOFF",
        },
      ],
      "short_name": "WALKOFF",
      "verbose_name": "",
    },
  ],
}
`
